resources/user-agents/apps/microsoft/bing-search-app.json

```diff
@@ -0,0 +1,201 @@
+            "HUAWEI TAG-L21": "Huawei TAG-L21",
+            "HUAWEI VNS-L31": "Huawei VNS-L31",
+            "JERRY": "Wiko Jerry",
+            "KIW-L21": "Huawei KIW-L21",
+            "K00Z": "Asus ME175CG",
+            "LENNY": "Wiko Lenny",
+            "Lenovo A616": "Lenovo A616",
+            "LG-D620": "LG D620",
+            "LG-H818": "LG H818",
+            "LIFETAB_P831X.2": "Medion LifeTab P831X.2",
+            "LIFETAB_P970X": "Lenovo LifeTab P970X",
+            "NEM-L51": "Huawei NEM-L51",
+            "P1040X": "Medion P1040X",
+            "SM-A300FU": "Samsung SM-A300FU",
+            "SM-A310F": "Samsung SM-A310F",
+            "SM-A500FU*": "Samsung SM-A500FU",
+            "SM-A510F": "Samsung SM-A510F",
+            "SM-G531F": "Samsung SM-G531F",
+            "SM-G550T": "Samsung SM-G550T",
+            "SM-G800F": "Samsung SM-G800F",
+            "SM-G850F*": "Samsung SM-G850F",
+            "SM-G900F": "Samsung SM-G900F",
+            "SM-G903F": "Samsung SM-G903F",
+            "SM-G925F*": "Samsung SM-G925F",
+            "SM-G930A": "Samsung SM-G930A",
+            "SM-G930F": "Samsung SM-G930F",
+            "SM-G935F": "Samsung SM-G935F",
+            "SM-J320F": "Samsung SM-J320F",
+            "SM-J510FN": "Samsung SM-J510FN",
+            "SM-N910V": "Samsung SM-N910V",
+            "SM-N920P": "Samsung SM-N920P",
+            "SM-N920V": "Samsung SM-N920V",
+            "SM-T235": "Samsung SM-T235",
+            "SM-T335": "Samsung SM-T335",
+            "SM-T525": "Samsung SM-T525",
+            "SM-T535": "Samsung SM-T535",
+            "SM-T555": "Samsung SM-T555",
+            "Tab2A7-10*": "Lenovo Tab 2 A7-10",
+            "VIE-L09": "Huawei VIE-L09",
+            "VK810 4G": "LG VK810",
+            "XT1068": "Motorola XT1068",
+            "ZTE Blade L5": "ZTE Blade L5"
+          },
+          "platforms": [
+            "Android_7_0",
+            "Android_6_0",
+            "Android_5_1", "Android_5_0",
+            "Android_4_4", "Android_4_3", "Android_4_2", "Android_4_1", "Android_4_0"
+          ]
+        },
+        {
+          "match": "Mozilla/5.0 (#PLATFORM#) applewebkit* (*khtml*like*gecko*) Version/* Chrome/* Mobile Safari/* BingWeb/#MAJORVER#.#MINORVER#*",
+          "platforms": [
+            "Android_7_0",
+            "Android_6_0",
+            "Android_5_1", "Android_5_0",
+            "Android_4_4", "Android_4_3", "Android_4_2", "Android_4_1", "Android_4_0",
+            "Android"
+          ]
+        },
+        {
+          "match": "Mozilla/5.0 (#PLATFORM#) applewebkit* (*khtml*like*gecko*) Chrome/* Mobile Safari/* BingWeb/#MAJORVER#.#MINORVER#*",
+          "platforms": [
+            "Android"
+          ]
+        },
+        {
+          "match": "Mozilla/5.0 (#PLATFORM##DEVICE# Build/*) applewebkit* (*khtml*like*gecko*) Version/* Mobile Safari/* BingWeb/#MAJORVER#.#MINORVER#*",
+          "engine": "WebKit",
+          "devices": {
+            "A700": "Acer A700",
```

Comment:
I think patterns could be reduced some here by not including devices that don't support certain android versions, for example, this device, the Acer Iconia Tab A700 seems to only support Android 4.0:

https://www.gsmarena.com/acer_iconia_tab_a700-4425.php
https://www.cnet.com/products/acer-iconia-tab-a700/specs/

And the "OnePlus A0001" from the Blink pattern above seems to only support Android 4.4 -> 6.0: https://en.wikipedia.org/wiki/OnePlus_One

Reply:
I splitted the file